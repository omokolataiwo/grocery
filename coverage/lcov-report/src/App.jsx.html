<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/App.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">src</a> App.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  Switch, Route, Redirect
} from 'react-router-dom';
import { library } from '@fortawesome/fontawesome-svg-core';
import {
  faTrash, faEdit, faCartPlus, faShoppingCart, faShoppingBag, faBriefcase, faChartLine, faHeart
} from '@fortawesome/free-solid-svg-icons';
import Groceries from './components/pages/Groceries';
import Home from './components/pages/Home';
import NavigationBar from './components/container/NavigationBar';
&nbsp;
library.add(faTrash, faEdit, faCartPlus, faShoppingCart, faBriefcase, faShoppingBag, faChartLine, faHeart);
&nbsp;
export class App extends Component {
  state = {};
&nbsp;
  render() {
    return (
      &lt;React.Fragment&gt;
        &lt;NavigationBar /&gt;
        &lt;Switch&gt;
          &lt;Route path="/home" component={Home} /&gt;
          &lt;Route path="/groceries" component={Groceries} /&gt;
          &lt;Redirect to="/home" /&gt;
        &lt;/Switch&gt;
      &lt;/React.Fragment&gt;
    );
  }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >()</span> =&gt; (<span class="cstat-no" title="statement not covered" >{})</span>;
export default connect(
  mapStateToProps,
  {}
)(App);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Dec 20 2018 18:53:15 GMT+0100 (West Africa Standard Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
