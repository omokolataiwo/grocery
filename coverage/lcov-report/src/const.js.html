<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/const.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">src</a> const.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11</td><td class="line-coverage quiet"><span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const BASE_URL = 'http://localhost:7000/api/v1';
export const FETCH_ALL_ITEMS = 'FETCH_ALL_ITEMS';
export const ADD_NEW_ITEM = 'ADD_NEW_ITEM';
export const ADD_NEW_ITEM_FAILURE = 'ADD_NEW_ITEM_FAILURE';
export const RESET_GROCERY_STATE = 'RESET_GROCERY_STATE';
export const DELETE_ITEM = 'DELETE_ITEM';
export const UPDATE_ITEM = 'UPDATE_ITEM';
export const UPDATE_ITEM_FAILURE = 'UPDATE_ITEM_FAILURE';
export const BUYING_ITEM = 'BUYING_ITEM';
export const BUYING_ITEM_ERROR = 'BUYING_ITEM_ERROR';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Dec 20 2018 17:51:30 GMT+0100 (West Africa Standard Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
