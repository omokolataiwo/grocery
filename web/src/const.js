export const BASE_URL = 'http://localhost:7000/api/v1';
export const FETCH_ALL_ITEMS = 'FETCH_ALL_ITEMS';
