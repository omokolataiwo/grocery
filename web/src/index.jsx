import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter, Route } from 'react-router-dom';
import App from './App';
import configureStore from './store';
import 'bootstrap/dist/css/bootstrap.min.css';

const store = configureStore();

ReactDOM.render(
  <Provider store={store}>
    <BrowserRouter>
      <Route path="/" component={App} />
    </BrowserRouter>
  </Provider>,
  document.getElementById('app')
);

if (module.hot) {
  module.hot.accept();
}
