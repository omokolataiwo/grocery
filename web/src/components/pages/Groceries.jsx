import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import GroceryItems from '../container/GroceryItems';
import AddGroceryItem from '../container/AddGroceryItem';
import {
  getAllItems, addItem, updateItem, resetGroceryState
} from '../../actions/grocery';
import { ADD_NEW_ITEM_FAILURE } from '../../const';

class Shopping extends Component {
  // static propTypes = {
  //   groceries: PropTypes.arrayOf(PropTypes.shape({
  //     id: PropTypes.string.isRequired,
  //     title: PropTypes.string.isRequired,
  //   })).isRequired
  // }

  state = {
    grocery: {
      name: '',
      amount: 0
    },
    errors: {
      name: [],
      amount: []
    }
  };

  componentDidMount() {
    const { getAllItems } = this.props;
    getAllItems();
  }

  componentDidUpdate() {
    const { type, error, resetGroceryState } = this.props;

    if (type === ADD_NEW_ITEM_FAILURE) {
      this.setState({ errors: error });
      resetGroceryState();
    }
  }

  onFormFieldChange = (event) => {
    const { name, value } = event.target;
    this.setState(prevState => ({
      grocery: { ...prevState.grocery, [name]: value }
    }));
  }

  onAddGrocery = (event) => {
    event.preventDefault();
    const { addItem, updateItem } = this.props;
    const { grocery, updating } = this.state;

    return !updating ? addItem(grocery) : updateItem(grocery);
  }

  render() {
    const { grocery, errors } = this.state;
    const { items } = this.props;

    return (
      <div>
        <h1>Grocery Listify</h1>
        <GroceryItems groceries={items || []} />
        <AddGroceryItem
          grocery={grocery}
          errors={errors}
          onChange={this.onFormFieldChange}
          addItem={this.onAddGrocery}
        />
      </div>
    );
  }
}

const mapStateToProps = ({ grocery: { type, items, error } }) => ({ type, items, error });
export default connect(
  mapStateToProps,
  {
    getAllItems, addItem, updateItem, resetGroceryState
  }
)(Shopping);
