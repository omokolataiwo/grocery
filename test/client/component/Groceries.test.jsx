import React from 'react';
import { shallow } from 'enzyme';
import { Groceries } from '../../../web/src/components/pages/Groceries';
import { ADD_NEW_ITEM_FAILURE, RESET_GROCERY_STATE } from '../../../web/src/const';

const props = {
  getAllItems: jest.fn(() => {}),
  resetGroceryState: jest.fn(() => {}),
  deleteItem: jest.fn(() => {}),
  addItem: jest.fn(() => {}),
  toggleBuy: jest.fn(() => {}),
  updateItem: jest.fn(() => {}),
  items: [
    {
      _id: 'msken',
      name: 'Milk',
      amount: '700',
      purchased: true
    },
    {
      _id: 'bken',
      name: 'Bread',
      amount: '300',
      purchased: false
    }
  ]
};
const wrapper = shallow(<Groceries {...props} />);

describe('Groceries Component', () => {
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when form field change', () => {
    const instance = wrapper.instance();
    instance.onFormFieldChange({ target: { name: 'name', value: 'Milk' } });
    instance.onFormFieldChange({ target: { name: 'amount', value: '300' } });
    expect(wrapper.state('grocery').name).toEqual('Milk');
    expect(wrapper.state('grocery').amount).toEqual('300');
  });

  it('should delete item', () => {
    const instance = wrapper.instance();
    instance.onDeleteItem(props.items[0]._id);
    expect(props.deleteItem).toHaveBeenCalledWith('msken');
  });

  it('should add item', () => {
    const instance = wrapper.instance();
    instance.onAddGrocery({ preventDefault: () => {} });
    expect(props.addItem).toHaveBeenCalledWith({ name: 'Milk', amount: '300', purchased: false });
  });

  it('should update item', () => {
    const instance = wrapper.instance();

    instance.onStartEditing({ name: 'Milk', amount: '300', purchased: false });
    instance.onAddGrocery({ preventDefault: () => {} });
    expect(props.updateItem).toHaveBeenCalledWith({
      name: 'Milk',
      amount: '300',
      purchased: false
    });
  });

  it('should clear item when back button is clicked', () => {
    const instance = wrapper.instance();

    instance.onStopEditing();
    expect(wrapper.state('grocery').name).toEqual('');
    expect(wrapper.state('isEditting')).toBeFalsy();
  });

  it('should buy and unbuy', () => {
    const instance = wrapper.instance();

    instance.toggleBuy('msken');
    expect(props.toggleBuy).toHaveBeenCalledWith('msken');
  });
});
