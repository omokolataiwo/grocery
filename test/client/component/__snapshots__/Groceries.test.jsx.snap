// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groceries Component should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Groceries
    addItem={[MockFunction]}
    deleteItem={[MockFunction]}
    getAllItems={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    items={
      Array [
        Object {
          "_id": "msken",
          "amount": "700",
          "name": "Milk",
          "purchased": true,
        },
        Object {
          "_id": "bken",
          "amount": "300",
          "name": "Bread",
          "purchased": false,
        },
      ]
    }
    resetGroceryState={[MockFunction]}
    toggleBuy={[MockFunction]}
    updateItem={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Grocery Listify
        </h1>,
        <div
          className="row"
        >
          <div
            className="items col-md-9"
          >
            <TransactionSummary
              items={
                Array [
                  Object {
                    "_id": "msken",
                    "amount": "700",
                    "name": "Milk",
                    "purchased": true,
                  },
                  Object {
                    "_id": "bken",
                    "amount": "300",
                    "name": "Bread",
                    "purchased": false,
                  },
                ]
              }
            />
            <GroceryItems
              deleteItem={[Function]}
              groceries={
                Array [
                  Object {
                    "_id": "msken",
                    "amount": "700",
                    "name": "Milk",
                    "purchased": true,
                  },
                  Object {
                    "_id": "bken",
                    "amount": "300",
                    "name": "Bread",
                    "purchased": false,
                  },
                ]
              }
              startEditing={[Function]}
              toggleBuy={[Function]}
            />
          </div>
          <div
            className="item-form col-md-3"
          >
            <AddGroceryItem
              errors={
                Object {
                  "amount": Array [],
                  "name": Array [],
                }
              }
              grocery={
                Object {
                  "amount": "",
                  "name": "",
                  "purchased": false,
                }
              }
              isEditing={false}
              onChange={[Function]}
              saveItem={[Function]}
              stopEditing={[Function]}
            />
          </div>
        </div>,
      ],
      "className": "main container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Grocery Listify",
        },
        "ref": null,
        "rendered": "Grocery Listify",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="items col-md-9"
            >
              <TransactionSummary
                items={
                  Array [
                    Object {
                      "_id": "msken",
                      "amount": "700",
                      "name": "Milk",
                      "purchased": true,
                    },
                    Object {
                      "_id": "bken",
                      "amount": "300",
                      "name": "Bread",
                      "purchased": false,
                    },
                  ]
                }
              />
              <GroceryItems
                deleteItem={[Function]}
                groceries={
                  Array [
                    Object {
                      "_id": "msken",
                      "amount": "700",
                      "name": "Milk",
                      "purchased": true,
                    },
                    Object {
                      "_id": "bken",
                      "amount": "300",
                      "name": "Bread",
                      "purchased": false,
                    },
                  ]
                }
                startEditing={[Function]}
                toggleBuy={[Function]}
              />
            </div>,
            <div
              className="item-form col-md-3"
            >
              <AddGroceryItem
                errors={
                  Object {
                    "amount": Array [],
                    "name": Array [],
                  }
                }
                grocery={
                  Object {
                    "amount": "",
                    "name": "",
                    "purchased": false,
                  }
                }
                isEditing={false}
                onChange={[Function]}
                saveItem={[Function]}
                stopEditing={[Function]}
              />
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TransactionSummary
                  items={
                    Array [
                      Object {
                        "_id": "msken",
                        "amount": "700",
                        "name": "Milk",
                        "purchased": true,
                      },
                      Object {
                        "_id": "bken",
                        "amount": "300",
                        "name": "Bread",
                        "purchased": false,
                      },
                    ]
                  }
                />,
                <GroceryItems
                  deleteItem={[Function]}
                  groceries={
                    Array [
                      Object {
                        "_id": "msken",
                        "amount": "700",
                        "name": "Milk",
                        "purchased": true,
                      },
                      Object {
                        "_id": "bken",
                        "amount": "300",
                        "name": "Bread",
                        "purchased": false,
                      },
                    ]
                  }
                  startEditing={[Function]}
                  toggleBuy={[Function]}
                />,
              ],
              "className": "items col-md-9",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "items": Array [
                    Object {
                      "_id": "msken",
                      "amount": "700",
                      "name": "Milk",
                      "purchased": true,
                    },
                    Object {
                      "_id": "bken",
                      "amount": "300",
                      "name": "Bread",
                      "purchased": false,
                    },
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "deleteItem": [Function],
                  "groceries": Array [
                    Object {
                      "_id": "msken",
                      "amount": "700",
                      "name": "Milk",
                      "purchased": true,
                    },
                    Object {
                      "_id": "bken",
                      "amount": "300",
                      "name": "Bread",
                      "purchased": false,
                    },
                  ],
                  "startEditing": [Function],
                  "toggleBuy": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <AddGroceryItem
                errors={
                  Object {
                    "amount": Array [],
                    "name": Array [],
                  }
                }
                grocery={
                  Object {
                    "amount": "",
                    "name": "",
                    "purchased": false,
                  }
                }
                isEditing={false}
                onChange={[Function]}
                saveItem={[Function]}
                stopEditing={[Function]}
              />,
              "className": "item-form col-md-3",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "errors": Object {
                  "amount": Array [],
                  "name": Array [],
                },
                "grocery": Object {
                  "amount": "",
                  "name": "",
                  "purchased": false,
                },
                "isEditing": false,
                "onChange": [Function],
                "saveItem": [Function],
                "stopEditing": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Grocery Listify
          </h1>,
          <div
            className="row"
          >
            <div
              className="items col-md-9"
            >
              <TransactionSummary
                items={
                  Array [
                    Object {
                      "_id": "msken",
                      "amount": "700",
                      "name": "Milk",
                      "purchased": true,
                    },
                    Object {
                      "_id": "bken",
                      "amount": "300",
                      "name": "Bread",
                      "purchased": false,
                    },
                  ]
                }
              />
              <GroceryItems
                deleteItem={[Function]}
                groceries={
                  Array [
                    Object {
                      "_id": "msken",
                      "amount": "700",
                      "name": "Milk",
                      "purchased": true,
                    },
                    Object {
                      "_id": "bken",
                      "amount": "300",
                      "name": "Bread",
                      "purchased": false,
                    },
                  ]
                }
                startEditing={[Function]}
                toggleBuy={[Function]}
              />
            </div>
            <div
              className="item-form col-md-3"
            >
              <AddGroceryItem
                errors={
                  Object {
                    "amount": Array [],
                    "name": Array [],
                  }
                }
                grocery={
                  Object {
                    "amount": "",
                    "name": "",
                    "purchased": false,
                  }
                }
                isEditing={false}
                onChange={[Function]}
                saveItem={[Function]}
                stopEditing={[Function]}
              />
            </div>
          </div>,
        ],
        "className": "main container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Grocery Listify",
          },
          "ref": null,
          "rendered": "Grocery Listify",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="items col-md-9"
              >
                <TransactionSummary
                  items={
                    Array [
                      Object {
                        "_id": "msken",
                        "amount": "700",
                        "name": "Milk",
                        "purchased": true,
                      },
                      Object {
                        "_id": "bken",
                        "amount": "300",
                        "name": "Bread",
                        "purchased": false,
                      },
                    ]
                  }
                />
                <GroceryItems
                  deleteItem={[Function]}
                  groceries={
                    Array [
                      Object {
                        "_id": "msken",
                        "amount": "700",
                        "name": "Milk",
                        "purchased": true,
                      },
                      Object {
                        "_id": "bken",
                        "amount": "300",
                        "name": "Bread",
                        "purchased": false,
                      },
                    ]
                  }
                  startEditing={[Function]}
                  toggleBuy={[Function]}
                />
              </div>,
              <div
                className="item-form col-md-3"
              >
                <AddGroceryItem
                  errors={
                    Object {
                      "amount": Array [],
                      "name": Array [],
                    }
                  }
                  grocery={
                    Object {
                      "amount": "",
                      "name": "",
                      "purchased": false,
                    }
                  }
                  isEditing={false}
                  onChange={[Function]}
                  saveItem={[Function]}
                  stopEditing={[Function]}
                />
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <TransactionSummary
                    items={
                      Array [
                        Object {
                          "_id": "msken",
                          "amount": "700",
                          "name": "Milk",
                          "purchased": true,
                        },
                        Object {
                          "_id": "bken",
                          "amount": "300",
                          "name": "Bread",
                          "purchased": false,
                        },
                      ]
                    }
                  />,
                  <GroceryItems
                    deleteItem={[Function]}
                    groceries={
                      Array [
                        Object {
                          "_id": "msken",
                          "amount": "700",
                          "name": "Milk",
                          "purchased": true,
                        },
                        Object {
                          "_id": "bken",
                          "amount": "300",
                          "name": "Bread",
                          "purchased": false,
                        },
                      ]
                    }
                    startEditing={[Function]}
                    toggleBuy={[Function]}
                  />,
                ],
                "className": "items col-md-9",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "items": Array [
                      Object {
                        "_id": "msken",
                        "amount": "700",
                        "name": "Milk",
                        "purchased": true,
                      },
                      Object {
                        "_id": "bken",
                        "amount": "300",
                        "name": "Bread",
                        "purchased": false,
                      },
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "deleteItem": [Function],
                    "groceries": Array [
                      Object {
                        "_id": "msken",
                        "amount": "700",
                        "name": "Milk",
                        "purchased": true,
                      },
                      Object {
                        "_id": "bken",
                        "amount": "300",
                        "name": "Bread",
                        "purchased": false,
                      },
                    ],
                    "startEditing": [Function],
                    "toggleBuy": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <AddGroceryItem
                  errors={
                    Object {
                      "amount": Array [],
                      "name": Array [],
                    }
                  }
                  grocery={
                    Object {
                      "amount": "",
                      "name": "",
                      "purchased": false,
                    }
                  }
                  isEditing={false}
                  onChange={[Function]}
                  saveItem={[Function]}
                  stopEditing={[Function]}
                />,
                "className": "item-form col-md-3",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "errors": Object {
                    "amount": Array [],
                    "name": Array [],
                  },
                  "grocery": Object {
                    "amount": "",
                    "name": "",
                    "purchased": false,
                  },
                  "isEditing": false,
                  "onChange": [Function],
                  "saveItem": [Function],
                  "stopEditing": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
